### ISSUE TITLE
Bundle error when using peerDependencies and devDependencies

### ISSUE THREAD

**Issue by jaydenseric** [ai/size-limit#83]
I spent an entire day and a half just trying to work out why size-limit was erroring 😥

It turns out, with this setup…

`index.mjs`:

```js
export { extractFiles } from 'extract-files'
```

`package.json`:

```json
{
  "dependencies": {
    "extract-files": "^5.0.0",
  },
  "devDependencies": {
    "size-limit": "^0.21.1"
  },
  "scripts": {
    "size": "size-limit"
  },
  "size-limit": [
    {
      "path": "index.mjs",
      "limit": "5 KB"
    }
  ]
}
```

`npm run size` works ok.

But if you change `package.json` to this:

```json
{
  "peerDependencies": {
    "extract-files": "^5.0.0"
  },
  "devDependencies": {
    "extract-files": "^5.0.0",
    "size-limit": "^0.21.1"
  },
  "scripts": {
    "size": "size-limit"
  },
  "size-limit": [
    {
      "path": "index.mjs",
      "limit": "5 KB"
    }
  ]
}
```

You get this error:

```
 ERROR  Error:
ERROR in ./index.mjs 1:0-44
Can't reexport the named export 'extractFiles' from non EcmaScript module (only default export is available)
 @ multi ./index.mjs
    at runWebpack.then.stats ([redacted]/node_modules/size-limit/index.js:214:15)
```

I suspect it might have something to do with peer dependencies being ignored:

https://github.com/ai/size-limit/blob/0.21.1/cli.js#L279

**Comment by ai** [comment]
Yeap seems like some sort of conflict between dependency ignore and ES modules.

Thanks for a good issue report. Can I ask you to help me in the investigation? (sorry, need to finish few things on the work before New Year).

1. Can you take `dependencies: { extract-files }` and add `size-limit.ignore: ["extract-files"]` to it? Let’s check that it’s related to ignore.
2. Can you try `"size-limit": "ai/size-limit#ignore-loader"` version. I added one possible fix.

**Comment by jaydenseric** [comment]
For (1), it had the same error. So definitely related.

For (2), with this `package.json`:

```json
{
  "dependencies": {
    "extract-files": "^5.0.0"
  },
  "devDependencies": {
    "size-limit": "ai/size-limit#ignore-loader"
  },
  "scripts": {
    "size": "size-limit"
  },
  "size-limit": [
    {
      "path": "index.mjs",
      "limit": "5 KB",
      "ignore": ["extract-files"]
    }
  ]
}
```

The error is:

```
 ERROR  Error:
ERROR in ./index.mjs 1:0-44
"export 'extractFiles' was not found in 'extract-files'
 @ multi ./index.mjs
    at runWebpack.then.stats ([redacted]/node_modules/size-limit/index.js:217:15)
```

**Comment by ai** [comment]
Do you have an idea how to change the webpack config to ignore named exports?

**Comment by ai** [comment]
If you have any ideas, you can try to add them here https://github.com/ai/size-limit/blob/master/index.js#L92-L93

Right now I have no idea how to deal with it :(

**Comment by bschlenk** [comment]
I'm running into a similar issue caused by using peerDependencies, but I get a different error `Unexpected character '@'`. I'd like to help out with this if I can. Does size-limit output the bundle it is analyzing somewhere? Or is there an easy way to make it do that? The error gives me a line number, but I have no idea how to see the code - being able to see it might help debug it.

**Comment by ai** [comment]
@bschlenk can you show your `package.json`?

**Comment by bschlenk** [comment]
I redacted some names, but this is the gist of it. I tried adding `"ignore": ["@stuff/my-other-package"]` and that didn't work. Everything is fine if I have `@stuff/my-other-package` under dependencies.

```
{
  "name": "my-package",
  "version": "1.0.0",
  "main": "dist/index.js",
  "types": "dist/index.d.ts",
  "scripts": {
    "test": "jest",
    "testWatch": "jest --watch",
    "prepublishOnly": "npm run genTypes && npm run compile && npm run size",
    "compile": "babel src/*.ts --extensions '.ts' --out-dir ./dist",
    "genTypes": "tsc -p build.tsconfig.json",
    "size": "size-limit"
  },
  "dependencies": {
    "@babel/runtime": "^7.2.0",
  },
  "peerDependencies": {
    "@stuff/my-other-package": ">= 2.0.0"
  },
  "devDependencies": {
    "@babel/cli": "^7.2.0",
    "@babel/core": "^7.2.0",
    "@babel/plugin-proposal-class-properties": "^7.1.0",
    "@babel/plugin-proposal-object-rest-spread": "^7.2.0",
    "@babel/plugin-syntax-dynamic-import": "^7.2.0",
    "@babel/plugin-transform-object-assign": "^7.2.0",
    "@babel/plugin-transform-runtime": "^7.2.0",
    "@babel/preset-env": "^7.2.0",
    "@babel/preset-typescript": "^7.1.0",
    "@babel/register": "^7.0.0",
    "@stuff/my-other-package": "^2.0.0",
    "@types/jest": "^23.3.10",
    "@types/node": "^10.12.12",
    "jest": "^23.6.0",
    "size-limit": "^0.21.0",
    "ts-jest": "^23.10.5",
    "typescript": "^2.7.2"
  },
  "size-limit": [
    {
      "path": "dist/index.js",
      "limit": "5 KB"
    }
  ]
}
```

**Comment by ai** [comment]
Thanks. I will look in next few days (sorry, I am preparing conference talk right now).

Can also you show the full error ouput?

**Comment by bschlenk** [comment]
```
 ERROR  Error:
ERROR in my-package from Terser
Unexpected character '@' [my-package/index.js:588,17]
    at getSize (/Users/bschlenk/my-package/node_modules/size-limit/index.js:240:15)
    at <anonymous>
```

**Comment by ai** [comment]
Fixed cdc608f

**Comment by ai** [comment]
Released in 1.3.3
